src/frontend/apps/lti_site/apps/bbb/components/UploadDocuments/index.tsx

```diff
@@ -0,0 +1,253 @@
+              )}
+              <Anchor
+                download
+                a11yTitle={intl.formatMessage(messages.downloadButtonLabel)}
+                label={intl.formatMessage(messages.downloadButtonLabel)}
+                style={{ fontFamily: 'Roboto-Medium' }}
+                title={intl.formatMessage(messages.downloadButtonLabel)}
+                href={url}
+              />
+            </Box>
+          ) : (
+            <Text>{uploadState}</Text>
+          )}
+        </Box>
+      )}
+    </Box>
+  );
+};
+
+export const UploadDocuments = () => {
+  const intl = useIntl();
+  const { data: classroomDocuments, refetch: refreshClassroomDocuments } =
+    useClassroomDocuments(bbbAppData.classroom!.id, {});
+
+  const { addUpload, uploadManagerState } = useUploadManager();
+  const [filesToUpload, setFilesToUpload] = useState<File[]>([]);
+  const [uploading, setUploading] = useState(false);
+
+  const uploadsInProgress = Object.values(uploadManagerState).filter((state) =>
+    [UploadManagerStatus.INIT, UploadManagerStatus.UPLOADING].includes(
+      state.status,
+    ),
+  );
+  const uploadsSucceeded = Object.values(uploadManagerState).filter((state) =>
+    [UploadManagerStatus.SUCCESS].includes(state.status),
+  );
+
+  const onDrop = (files: any) => {
+    setFilesToUpload(filesToUpload.concat(files));
+  };
+
+  const uploadFiles = () => {
+    const file = filesToUpload.shift();
+    if (!file) {
+      return;
+    }
+
+    setUploading(true);
+    createClassroomDocument({
+      filename: file.name,
+      size: file.size,
+    }).then((response) => {
+      addUpload(modelName.CLASSROOM_DOCUMENTS, response.id, file);
+      refreshClassroomDocuments();
+    });
+  };
+
+  useEffect(() => {
+    if (uploading) {
+      if (filesToUpload.length > 0) {
+        uploadFiles();
+      } else {
+        setUploading(false);
+      }
+    }
+  }, [uploadsSucceeded.length]);
+
+  window.addEventListener('classroomDocumentUpdated', () => {
+    refreshClassroomDocuments();
+  });
+
+  return (
+    <Box>
+      <Box
+        align="center"
+        background="blue-message"
+        fill
+        border={{
+          style: filesToUpload.length === 0 ? 'dashed' : 'solid',
+          size: 'xsmall',
+          color: 'blue-active',
+        }}
+        margin={{ top: 'medium' }}
+        round="xsmall"
+      >
+        {classroomDocuments?.results.map((classroomDocument) => (
+          <UploadDocumentsRow
+            key={classroomDocument.id}
+            filename={classroomDocument.filename}
+            isDefault={classroomDocument.is_default}
+            documentId={classroomDocument.id}
+            // setDefaultDocument={setDefaultDocument(classroomDocument)}
+            uploadState={classroomDocument.upload_state}
+            url={classroomDocument.url}
+          />
+        ))}
+
+        {uploadsInProgress.map((state) => (
+          <UploadDocumentsRow
+            key={state.file.name}
```

Comment:
Are we sure file name is unique ?

Reply:
i replaced the key with `{`${state.file.name}_${index}`}`
WDYT ?